### ISSUE TITLE
Lookbehind in regular expressions not reported even though it is not supported on Safari browser

### ISSUE THREAD

**Issue by Zhouqchao** [amilajack/eslint-plugin-compat#555]
`lookbehind` in regular expresions should report an error because it is not supported on Safari browser.
![image](https://user-images.githubusercontent.com/24250131/227114423-99a5b0b3-f0bc-4b5f-9091-626637f2ea91.png)

here is my browserslist config: 
```json
{
	"browserslist": [
	    "last 2 years",
	    "not dead"
	  ]
}
```

I do get an error for OffscreenCanvas not being supported on Safari 14, so it looks like I have set up the plugin correctly.
![image](https://user-images.githubusercontent.com/24250131/227113208-0563faf8-ce2b-4915-a759-56519e6688a5.png)

**Comment by RobinBlomberg** [comment]
For a basic workaround:
```js
rules: {
  "no-restricted-syntax": [
    "error",
    {
      selector: "Literal[regex][raw=/\\(\\?<[=!]/]",
      message:
        "Lookbehind assertions are not supported in some browsers (e.g. Safari 16.3).",
    },
  ],
},
```
